plugin-tray/xcbutils.h

```diff
@@ -53,13 +52,19 @@ namespace Xcb
 {
 typedef xcb_window_t WindowId;
 
+struct ScopedCPointerDeleter
+{
+    static inline void cleanup(void *pointer) noexcept { free(pointer); }
+    void operator()(void *pointer) const noexcept { cleanup(pointer); }
+};
+
```

Comment:
@palinek, please also take a look at this (your code again).

Reply:
What about this?

Reply:
I took it from Qt by the way: [qscopedpointer.h#L51](https://github.com/qt/qtbase/blob/ceeaf1b65787d3ef46dd0bdbd0e6810137163fea/src/corelib/tools/qscopedpointer.h#L51)

Reply:
Actually `cleanup()` seems to be called only by `QScoperPointer` and since we ported code to `std::unique_ptr` we can get rid of it and free directly.

Reply:
I had no problem with this one; just asked @palinek to take a look at it, for the sake of certainty.